src/core_plugins/kibana/public/dashboard/directives/dashboard_panel.js

```diff
@@ -1,15 +1,19 @@
 import _ from 'lodash';
 import 'ui/visualize';
 import 'ui/doc_table';
-import { loadPanelProvider } from 'plugins/kibana/dashboard/components/panel/lib/load_panel';
+import 'plugins/kibana/dashboard/services/get_object_loaders_for_dashboard.factory';
 import FilterManagerProvider from 'ui/filter_manager';
 import uiModules from 'ui/modules';
 import panelTemplate from 'plugins/kibana/dashboard/components/panel/panel.html';
+import { getPersistedStateId } from 'plugins/kibana/dashboard/components/panel/lib/panel_state';
+import { loadSavedObject } from 'plugins/kibana/dashboard/components/panel/lib/load_saved_object';
 
 uiModules
 .get('app/dashboard')
-.directive('dashboardPanel', function (savedVisualizations, savedSearches, Notifier, Private, $injector) {
-  const loadPanel = Private(loadPanelProvider);
+.directive('dashboardPanel', function ($injector) {
+  const Private = $injector.get('Private');
```

Comment:
> That is still tightly coupled to Angular's DI

Everything will be, you can't use Angular without the DI 😢 .

There's actually 2 things going on here though. First is the removal of `Private`, which I think is universal desire.

The second is the use of `$injector` to make it clear this is Angular stuff, so allow the code to be minified without duplicating all the dependencies, and I guess tangentially to make the code still look like the `Private` code. This change is very much up for debate, and was just an idea that was thrown out when discussing the Angular styleguide (which is also kind of in flux). Sounds like folks aren't too keen on this idea, so maybe it's best to remove it for now and discuss it outside this PR.

> I think the idea is to use $injector only for nameless functions, not for things already registered with angular. 

Actually, `$injector` only works for named, registered Angular code. It doesn't work the same way the `Private` does.